From 81e6602984c40bb7af973412a6ae3036c511eddb Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Thu, 2 Jan 2014 13:13:19 +0000
Subject: Inhibit -wmac warnings due to pic and eqn

Define .PS and .PE to empty strings in pic output to inhibit -wmac
warnings; similarly, define .EQ and .EN to empty strings in eqn output.

Bug-Debian: http://bugs.debian.org/495713
Forwarded: no
Last-Update: 2013-01-07

Patch-Name: pic-eqn-warnings.patch
---
 src/preproc/eqn/main.cpp | 2 ++
 src/preproc/pic/main.cpp | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/src/preproc/eqn/main.cpp b/src/preproc/eqn/main.cpp
index 57347f13..52615c47 100644
--- a/src/preproc/eqn/main.cpp
+++ b/src/preproc/eqn/main.cpp
@@ -373,6 +373,8 @@ int main(int argc, char **argv)
     }
   init_table(device);
   init_char_table();
+  printf(".if !dEQ .ds EQ\n"
+	 ".if !dEN .ds EN\n");
   if (output_format == troff) {
     printf(".if !'\\*(.T'%s' "
 	   ".if !'\\*(.T'html' "	// the html device uses `-Tps' to render
diff --git a/src/preproc/pic/main.cpp b/src/preproc/pic/main.cpp
index 19065a66..236121d5 100644
--- a/src/preproc/pic/main.cpp
+++ b/src/preproc/pic/main.cpp
@@ -615,7 +615,11 @@ int main(int argc, char **argv)
     out = make_fig_output();
   else
 #endif
+  {
     out = make_troff_output();
+    printf(".if !dPS .ds PS\n"
+	   ".if !dPE .ds PE\n");
+  }
 #ifdef FIG_SUPPORT
   if (whole_file_flag) {
     if (optind >= argc)
